src/xrGame/screenshots_writer.cpp

```diff
@@ -33,7 +33,7 @@ void writer::set_player_cdkey_digest(shared_str const& cdkeydigest)
     m_info_data.w_string(ss_info_secion, ss_admin_name_key, admin_name.c_str());
 }*/
 
-u32 const writer::write_info(sha_process_yielder* yielder)
+u32 const writer::write_info(crypto::yielder_t* yielder)
```

Comment:
weak_ptr.

Reply:
No, weak_ptr requires shared_ptr to own the yielder. shared_ptr has big runtime.
It is better to be sure that yielder will never outlives its caller. As far as writer::write_info calls in the same thread as its caller, we are safe.

Maybe you would like something like this:
```suggestion
u32 const writer::write_info(const crypto::yielder_t& yielder)
```